apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -464,6 +468,20 @@ type BorrowWithinCohort struct {
 	MaxPriorityThreshold *int32 `json:"maxPriorityThreshold,omitempty"`
 }
 
+// FairSharing contains the properties of the ClusterQueue when participating in fair sharing.
+type FairSharing struct {
+	// weight gives a comparative advantage to this ClusterQueue when competing for unused
+	// resources in the cohort against other ClusterQueues.
+	// The share of a ClusterQueue is based on the dominant resource usage above nominal
+	// quotas for each resource, divided by the weight.
+	// Admission prioritizes scheduling workloads from ClusterQueues with the lowest share
+	// and preempting workloads from the ClusterQueues with the highest share.
+	// A zero weight implies infinite share value, meaning that this ClusterQueue will always
+	// be at disadvantage against other ClusterQueues.
+	// +kubebuilder:default=1
+	Weight *resource.Quantity `json:"weight,omitempty"`
```

Comment:
Can't `*int32` fit here? Or `IntOrString`? What other values we'll have?

Reply:
The quantity allows you to have integers or decimals, without losing precision.